### ISSUE TITLE
Automatize publication on F-Droid

### ISSUE THREAD

**Issue by kelson42** [kiwix/kiwix-android#1682]
Doing this manually is too complicated.

**Comment by macgills** [comment]
@kelson42 the output from [this discussion](https://gitlab.com/fdroid/fdroiddata/issues/1823) is that we need to host a file like [this](https://ota.koreader.rocks/koreader-android-fdroid-latest) that has our versionName/Code and update it on every tag/release.
We will also need to update our fdroid metadata to [UpdateCheckMode-Http](https://f-droid.org/fr/docs/Build_Metadata_Reference/#UpdateCheckMode) 
We will also require fdroid to support build.gradle.kts files, ticket [here](https://gitlab.com/fdroid/fdroidserver/issues/613)

We would need to decide where to host this file, download.kiwix the build process is already able to upload to but that may not be the best place

**Comment by kelson42** [comment]
@macgills This file should be generated at release time by the gradle and put somewhere. Maybe in the Github wiki (should be downloadable as a raw text)?! But I'm not wanting you to invest any time on this as long as FDroid is not 100% ready.

Do we have a timeline for the implementation of build.gradle.kts support within FDroid?
Any chance to make a manual publication in the meantime?

**Comment by kelson42** [comment]
Blocked by https://gitlab.com/fdroid/fdroidserver/issues/613

**Comment by kelson42** [comment]
@macgills https://gitlab.com/fdroid/fdroidserver/-/issues/613 has been implemented 3 months ago. Do we have any blocker left here?

**Comment by macgills** [comment]
I would guess that it has been unblocked

**Comment by kelson42** [comment]
@macgills Great, considering that this is near-impossible to get our PR merged manually, we should definitly move on with this early in 3.5 milestone and make a release (3.4.x).

**Comment by licaon-kter** [comment]
Where did the 7 come from? https://gitlab.com/futuretechnoworld.mohit/kiwix-fdroid/-/jobs/2526061190#L582

`Could not build app org.kiwix.kiwixmobile: Unexpected version/version code in output; APK: '3.5.0' / '7230500',  Expected: '3.5.0' / '230500'`

**Comment by licaon-kter** [comment]
Already done

**Comment by licaon-kter** [comment]
> Such an automatiion would imply to do automaticaly (1) an account in gitlab (2) a place to make a fork (3) create a MR and make a commit automatically (4) push the code (5) opena MR (6) what do you do if a change is requested or a problem occrus?....

_That's not the automation you are looking for..._

Put versionCode & versionName in some file that we can grep and tag after that. Done. ;)

**Comment by kelson42** [comment]
@licaon-kter And Fdroid will have something checking this periodically? You have an example of an app working like this?

**Comment by MohitMaliFtechiz** [comment]
hi @licaon-kter ,

So basically we need to host file which pointed to the latest versioncode , versionname and tag , similar like https://ota.koreader.rocks/koreader-android-fdroid-latest is that correct  ?

@kelson42 how we can host this file on download.kiwix.org?

**Comment by licaon-kter** [comment]
@MohitMaliFtechiz versionName and Code only, and have versionCode match Tag

**Comment by MohitMaliFtechiz** [comment]
> @MohitMaliFtechiz versionName and Code only, and have versionCode match Tag

Hi @licaon-kter one more question can we include file on our git repo with this information will that work too ?

**Comment by kelson42** [comment]
@MohitMaliFtechiz I would do it directly in git. OK?

**Comment by licaon-kter** [comment]
Yes :)

**Comment by MohitMaliFtechiz** [comment]
> @MohitMaliFtechiz I would do it directly in git. OK?



> @MohitMaliFtechiz I would do it directly in git. OK?

ok

**Comment by MohitMaliFtechiz** [comment]
hi @licaon-kter ,

I have just issue the PR with file containing the information can you please check once.

Thanks and Regards
Mohit Mali

**Comment by licaon-kter** [comment]
@MohitMaliFtechiz typo aside (it's spelled F-Droid :stuck_out_tongue: ) https://github.com/kiwix/kiwix-android/pull/2885 looks good
 
A tentative recipe modification would be:
```
...
AutoUpdateMode: Version
UpdateCheckMode: Tags
UpdateCheckData: RELEASE_INFO|\s([0-9]*)||
CurrentVersion: 3.5.0
CurrentVersionCode: 7230500
```

Ref: https://f-droid.org/docs/Build_Metadata_Reference/#UpdateCheckData

But we can be sure everything is fine after you (merge of course then) update it for next version and then tag your next version.

**Comment by kelson42** [comment]
@licaon-kter File to catch latest version info will be available at https://download.kiwix.org/release/kiwix-android/VERSION_INFO

**Comment by licaon-kter** [comment]
We were lost in translation a bit :)

@kelson42 in my post above, it's just a bit of the recipe, the end of it.

That `UpdateCheckData` line takes the file named `RELEASE_INFO`, parses it and extracts the needed info.

At the moment when I posted that, the RELEASE_INFO file from your PR looked like [this](https://github.com/kiwix/kiwix-android/pull/2885/commits/baa054a3b99789dbf9687428bb3a77d28463470f):
```
3.5.0
7230500
```
...and the regex is FOR THIS FILE FORMAT only.

**Comment by licaon-kter** [comment]
https://gitlab.com/fdroid/fdroiddata/-/commit/d764273668aa1cce93d712f444bfe319db31c5d0

Keep this open until we can check its correctness ;)

**Comment by kelson42** [comment]
@licaon-kter Thx, so moving this ticket to the next release to check if 3.6.0 will be automatically update on F-Droid.

**Comment by MohitMaliFtechiz** [comment]
hi @licaon-kter , We have just released 3.6.0 , version information is hosted on https://download.kiwix.org/release/kiwix-android/VERSION_INFO. Please proceed with auto publish on Fdroid.

**Comment by licaon-kter** [comment]
@MohitMaliFtechiz tested locally and updated autoupdate mode of operation: https://gitlab.com/fdroid/fdroiddata/-/commit/6417334e120538e5018bf81734826d099bf2eb16

It should be picked up in 12-20h

**Comment by licaon-kter** [comment]
Simplified the check a bit to use repo hosted file instead ( eg: https://github.com/kiwix/kiwix-android/blob/3.6.0/VERSION_INFO ), so updates are picked up only when tagging: https://gitlab.com/fdroid/fdroiddata/-/commit/5e27e7038da26343a94c031ceb929cad243f4c68

**Comment by licaon-kter** [comment]
Success https://gitlab.com/fdroid/fdroiddata/-/commit/32b7a899499e1ab75195b5ec9a00cb68ea37f4f2 

/close this

**Comment by kelson42** [comment]
@licaon-kter @MohitMaliFtechiz I don't think this last move was appropriate because with the current system the `VERSION_INFO` is generated by the Github Workflow but not commited to the repo. This would be an operation to happen manually. We have been fooled I believe because @MohitMaliFtechiz has commited/pushed that file to the git repository. But actually it should only live in https://download.kiwix.org/release/kiwix-android/. Right?

**Comment by MohitMaliFtechiz** [comment]
hi @licaon-kter @kelson42 , 

Yes we should use https://download.kiwix.org/release/kiwix-android/VERSION_INFO this is because this will auto uploaded during release via gradle task  ```generateVersionCodeAndName```  , while on git hub during creating final PR release i am running ```generateVersionCodeAndName``` on local and this changes VERSION_INFO were committed along other.

**Comment by licaon-kter** [comment]
@MohitMaliFtechiz done https://gitlab.com/fdroid/fdroiddata/-/commit/dca017aaba6bbd040e609a28d2563d68c840c215
